### PR TITLE
Allow use of just image name without the tag

### PR DESCRIPTION
**Pull request by brahmaroutu**
Issue with filtering by image name is mentioned in #10645 (which is about filtering by container name)

Signed-off-by: Srini Brahmaroutu srbrahma@us.ibm.com

### CODE REVIEW

**Comment by tiborvass** [comment]
Related PRs: #10663 #10646

@brahmaroutu can you please update the description to explain what this PR does, especially compared to those two PRs?

**Comment by brahmaroutu** [comment]
PR #10646 is fixing the bug and should be merged first, "User should be able to use shorted form of container id in the filter"
PR #10663 is a enhancement to the filtering logic. "User should be able to filter event using container name"
PR #10675 "Allow user to filter events using image name without the tag, ex: busybox instead of busybox:latest"

**Comment by tiborvass** [comment]
@brahmaroutu on a second thought, I really don't see why not fixing this. There is no container/image ambiguity because of the `image=` prefix. I'm pushing this to code review.

**Comment by brahmaroutu** [comment]
@tiborvass Yes, we should fix this issue. Thanks.

File: integration-cli/docker_cli_events_test.go [hunk 1]
```diff
@@ -296,3 +296,69 @@ func TestEventsFilters(t *testing.T) {
...
+	}
+	container1 := stripTrailingCharacters(out)
+	out, _, err = runCommandWithOutput(exec.Command(dockerBinary, "run", "--name", "container_2", "-d", "busybox", "true"))
+	if err != nil {
+		t.Fatal(out, err)
+	}
+	container2 := stripTrailingCharacters(out)
+
+	eventsCmd := exec.Command(dockerBinary, "events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", time.Now().Unix()), "--filter", "image=busybox:latest")
+	out, exitCode, err := runCommandWithOutput(eventsCmd)
```
Comment (tiborvass) [inline integration-cli/docker_cli_events_test.go]: no need for exitCode

File: integration-cli/docker_cli_events_test.go [hunk 2]
```diff
@@ -296,3 +296,69 @@ func TestEventsFilters(t *testing.T) {
...
+			count1++
+		} else if strings.Contains(e, container2) {
+			count2++
+		}
+	}
+	if count1 == 0 || count2 == 0 {
+		t.Fatalf("Expected events from each container but got %d from %s and %d from %d", count1, container1, count2, container2)
+	}
+
+	eventsCmd = exec.Command(dockerBinary, "events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", time.Now().Unix()), "--filter", "image=busybox")
```
Comment (tiborvass) [inline integration-cli/docker_cli_events_test.go]: can you reuse the code above in a for loop where you iterate in `[]string{"busybox:latest", "busybox"}` ?

File: integration-cli/docker_cli_events_test.go [hunk 3]
```diff
@@ -296,3 +296,69 @@ func TestEventsFilters(t *testing.T) {
 
 	logDone("events - filters")
 }
+
+func TestEventsFilterContainerName(t *testing.T) {
```
Comment (tiborvass) [inline integration-cli/docker_cli_events_test.go]: this should say `ImageName`

File: integration-cli/docker_cli_events_test.go [hunk 4]
```diff
@@ -296,3 +296,69 @@ func TestEventsFilters(t *testing.T) {
...
+	count2 := 0
+	for _, e := range events {
+		if strings.Contains(e, container1) {
+			count1++
+		} else if strings.Contains(e, container2) {
+			count2++
+		}
+	}
+	if count1 == 0 || count2 == 0 {
+		t.Fatalf("Expected events from each container but got %d from %s and %d from %d", count1, container1, count2, container2)
```
Comment (tiborvass) [inline integration-cli/docker_cli_events_test.go]: last element should be `%s` not `%d`.

**Comment by brahmaroutu** [comment]
@tiborvass the tests are updated

**Comment by tiborvass** [comment]
LGTM

Ping @LK4D4 @estesp merge when you think it's good

**Comment by estesp** [comment]
LGTM

**Pull request merged**